common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,60 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
-  }
+function buildUrl({ sortKey }, query) {
+  return queryString.stringify({
+    ...query,
+    sortBy: sortKey,
+    sortDirection: query.sortDirection === 'asc' ? 'desc' : 'asc',
+  })
+}
 
-  return pickBy({
-    ...head,
-    html: head.html ? i18n.t(head.html) : undefined,
-    text: head.text ? i18n.t(head.text) : undefined,
+function buildAriaLabel(head, { sortDirection }) {
+  // the label indicates the result of the sorting action triggered by the link,
+  // which is the opposite of the current state
+  return i18n.t('sort', {
+    context: sortDirection === 'asc' ? 'descending' : 'ascending',
+    label: i18n.t(head.html).toLowerCase(),
   })
 }
 
+function buildWrapper(head, query) {
+  const label = i18n.t(head.html)
+  if (!head.isSortable) {
+    return label
+  }
+  const url = buildUrl(head, query)
+  const ariaLabel = buildAriaLabel(head, query)
+  return `<a aria-label="${ariaLabel}" class="sortable-table__button" role="button" href="?${url}">${label}</a>`
+}
+
+function assignSortAttributes(head, { sortBy, sortDirection }) {
+  if (!head.isSortable) {
+    return head.attributes ? { ...head.attributes } : undefined
```

Comment:
I don't think you need to use spread if the attributes already exist here?

Wouldn't this be enough:

```javascript
return head.attributes ? head.attributes : undefined
```

And I think if you destructure head in the argument this could read even nicer:

```javascript
function assignSortAttributes({ attributes, isSortable, sortKey }, { sortBy, sortDirection }) {
  if (!isSortable) {
    return attributes
  }
  
  const isCurrentSortColumn = sortKey === sortBy
  const ariaSortDirection = sortDirection === 'asc' ? 'ascending' : 'descending'

  return {
    ...attributes,
    'aria-sort': isCurrentSortColumn ? ariaSortDirection : 'none',
  }
}
```

I think it also helps to be more consistent with the destructuring of the arguments rather than have a mix of patterns (unless you need to return `head` in its entirety, which is doesn't seem like you do in this case).